x/multisig/msg.go

```diff
@@ -44,19 +36,62 @@ func (UpdateContractMsg) Path() string {
 
 // Validate enforces sigs and threshold boundaries
 func (c *UpdateContractMsg) Validate() error {
-	if len(c.Sigs) == 0 {
-		return errors.Wrap(errors.ErrInvalidMsg, "missing sigs")
+	switch n := len(c.Participants); {
+	case n == 0:
+		return errors.Wrap(errors.ErrInvalidMsg, "no participants")
+	case n > 100:
+		return errors.Wrap(errors.ErrInvalidMsg, "too many participants")
 	}
-	if c.ActivationThreshold <= 0 || int(c.ActivationThreshold) > len(c.Sigs) {
-		return errors.Wrap(errors.ErrInvalidMsg, invalidThreshold)
-	}
-	if c.AdminThreshold <= 0 {
-		return errors.Wrap(errors.ErrInvalidMsg, invalidThreshold)
+	return validateWeights(errors.ErrInvalidMsg,
+		c.Participants, c.ActivationThreshold, c.AdminThreshold)
+}
+
+// validateWeights returns an error if given participants and thresholds
```

Comment:
Very nice and thorough set of checks.
I know the validate functions are tedious, but I think it is very important to be meticulous here, as they cut off many attack vectors as well as user errors.